packages/block/template.ts

```diff
@@ -0,0 +1,153 @@
+import { AbstractBlock } from './types';
+import { setHas$ } from './dom';
+import type { Edit, VElement , Hole } from './types';
+
+const X_CHAR = 120;
+const VOID_ELEMENTS = new Set([
+  'area',
+  'base',
+  'basefont',
+  'bgsound',
+  'br',
+  'col',
+  'command',
+  'embed',
+  'frame',
+  'hr',
+  'image',
+  'img',
+  'input',
+  'isindex',
+  'keygen',
+  'link',
+  'menuitem',
+  'meta',
+  'nextid',
+  'param',
+  'source',
+  'track',
+  'wbr',
+]);
+
+export const renderToTemplate = (
+  vnode: VElement,
+  edits: Edit[] = [],
+  path: number[] = [],
+): string => {
+  let props = '';
+  let children = '';
+  const current: Edit = {
+    path, // The location of the edit in in the virtual node tree
+    edits: [], // Occur on mount + patch
+    inits: [], // Occur before mount
+  };
+
+  for (let name in vnode.props) {
+    const value = vnode.props[name];
+    if (name === 'key' || name === 'ref' || name === 'children') {
+      continue;
+    }
+    if (name === 'className') name = 'class';
+    if (name === 'htmlFor') name = 'for';
+
+    if (name.startsWith('on')) {
+      const isValueHole = '__key' in value;
+      // Make objects monomorphic
+      current.edits.push({
+        type: 'event',
+        listener: isValueHole ? value.key : value,
+        name,
+        hole: isValueHole ? value.key : undefined,
+        value: undefined,
+        index: undefined,
+        patch: undefined,
+        block: undefined,
+      });
+
+      continue;
+    }
+
+    if ('__key' in value) {
```

Comment:
Still having the same bug on this line. Checking if the value is an object before does solve the issue:

```suggestion
    if (typeof value === 'object' && '__key' in value) {
```

<img width="208" alt="Screenshot 2023-02-25 at 21 38 04" src="https://user-images.githubusercontent.com/43268759/221378579-843f80fa-80a0-4543-875f-55cc20931f4b.png">